diff --git i/src/storage/mysql_catalog.cpp w/src/storage/mysql_catalog.cpp
index 2c993cc..0bd4fc6 100644
--- i/src/storage/mysql_catalog.cpp
+++ w/src/storage/mysql_catalog.cpp
@@ -30,7 +30,7 @@ MySQLCatalog::MySQLCatalog(AttachedDatabase &db_p, string connection_string_p, s
 
 MySQLCatalog::~MySQLCatalog() = default;
 
-string EscapeConnectionString(const string &input) {
+static string EscapeConnectionString(const string &input) {
 	string result = "\"";
 	for (auto c : input) {
 		if (c == '\\') {
@@ -64,7 +64,7 @@ string AddConnectionOption(const KeyValueSecret &kv_secret, const string &name,
 	return result;
 }
 
-unique_ptr<SecretEntry> GetSecret(ClientContext &context, const string &secret_name) {
+static unique_ptr<SecretEntry> GetSecret(ClientContext &context, const string &secret_name) {
 	auto &secret_manager = SecretManager::Get(context);
 	auto transaction = CatalogTransaction::GetSystemCatalogTransaction(context);
 	// FIXME: this should be adjusted once the `GetSecretByName` API supports this
diff --git i/src/storage/mysql_schema_entry.cpp w/src/storage/mysql_schema_entry.cpp
index 889ff75..890013a 100644
--- i/src/storage/mysql_schema_entry.cpp
+++ w/src/storage/mysql_schema_entry.cpp
@@ -171,7 +171,7 @@ void MySQLSchemaEntry::Alter(CatalogTransaction transaction, AlterInfo &info) {
 	tables.AlterTable(transaction.GetContext(), alter);
 }
 
-bool CatalogTypeIsSupported(CatalogType type) {
+static bool CatalogTypeIsSupported(CatalogType type) {
 	switch (type) {
 	case CatalogType::INDEX_ENTRY:
 	case CatalogType::TABLE_ENTRY:
